core/nakamoto/mempool.go

```diff
@@ -19,16 +30,67 @@ type FeeRates struct {
 
 // NewMempool creates a new mempool.
 func NewMempool() *Mempool {
-	return &Mempool{}
+	return &Mempool{
+		candidateTxns: []*RawTransaction{},
+		confirmedTxns: []*Transaction{},
+		fees: &FeeRates{
+			MinFee:    math.MaxUint64,
+			MedianFee: 0,
+			MaxFee:    0,
+		},
+	}
+}
+
+func (m *Mempool) insertCandidate(tx *RawTransaction) {
```

Comment:
So basically this inserts a new tx into the candidate set, which is a list of txs in memory ordered by fee descending? And then later we filter this down to produce a bundle, right?